okhttp-client/src/main/scala/org/http4s/client/okhttp/OkHttp.scala

```diff
@@ -0,0 +1,140 @@
+package org.http4s.client.okhttp
+
+import java.io.IOException
+
+import cats.data._
+import cats.effect._
+import cats.effect.implicits._
+import okhttp3.{
+  Call,
+  Callback,
+  OkHttpClient,
+  Protocol,
+  RequestBody,
+  Headers => OKHeaders,
+  MediaType => OKMediaType,
+  Request => OKRequest,
+  Response => OKResponse
+}
+import okio.BufferedSink
+import org.http4s.{Header, Headers, HttpVersion, Method, Request, Response, Status}
+import org.http4s.client.{Client, DisposableResponse}
+import fs2.Stream._
+import fs2._
+import fs2.io._
+
+import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext
+
+object OkHttp {
+
+  def defaultconfig[F[_]]()(implicit F: Effect[F]): F[OkHttpClient.Builder] = F.delay {
+    new OkHttpClient.Builder()
+      .protocols(
+        List(
+          Protocol.HTTP_2,
+          Protocol.HTTP_1_1
+        ).asJava)
+  }
+
+
+  def apply[F[_]](config: F[OkHttpClient.Builder] = null)(
```

Comment:
I might prefer `default` and `apply` that takes a config instead of a null-default.  Even though I recognize what you're trying to do and it's safe, I flinched when I saw the null.

Does `build()` mutate the builder?  If not, I'm not sure why it needs to be passed in in `F`.

Reply:
I originally wanted to do `config: F[OkHttpClient.Builder] = defautconfig[F]` but that's not valid, so this was my solution, I could split it into two methods to remove the null, as in the above comment, the `Builder` has a lot of nasty state in it.

Reply:
Yeah, that implicit not being visible to default arguments is a drag.

Let's see if anyone else chimes in on the design here.

Reply:
We have it split in two methods in a lot of other places. For consistency, I would like that here as well, but I'm not very opinionated on it.